pkg/validator/validators.go

```diff
@@ -145,16 +146,39 @@ func egressValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionR
 		return nil, err
 	}
 
+	// Validate match references
+	allowedAPIGroups := []string{smiSpecs.SchemeGroupVersion.String(), policyv1alpha1.SchemeGroupVersion.String()}
+	upstreamTrafficSettingMatchCount := 0
 	for _, m := range egress.Spec.Matches {
-		if m.Kind != "HTTPRouteGroup" {
-			return nil, errors.Errorf("Expected 'Matches.Kind' to be 'HTTPRouteGroup', got: %s", m.Kind)
-		}
+		switch *m.APIGroup {
+		case smiSpecs.SchemeGroupVersion.String():
+			switch m.Kind {
+			case "HTTPRouteGroup":
+				// no additional validation
+
+			default:
+				return nil, errors.Errorf("Expected 'Matches.Kind' for match '%s' to be 'HTTPRouteGroup', got: %s", m.Name, m.Kind)
```

Comment:
To match the messages we already have, let's keep using the field names as they appear in the API vs. the Go struct field names:
```suggestion
				return nil, errors.Errorf("Expected 'matches.kind' for match '%s' to be 'HTTPRouteGroup', got: %s", m.Name, m.Kind)
```

This might also be a bit more clear if it reads something more like `Expected spec.matches[i].kind to be ...` with the full path in the YAML/JSON to exactly what's not valid.